<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 6.0.0">


  <link rel="apple-touch-icon" sizes="180x180" href="/images/favicon.jpeg">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon.jpeg">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon.jpeg">
  <link rel="mask-icon" href="/images/favicon.jpeg" color="#222">
  <meta name="google-site-verification" content="k6DpQ18iYt6OkDbAzpaogHDiffDlR-MP0FIFP9M1dXo">
  <meta name="baidu-site-verification" content="code-Hpn6nthL0z">

<link rel="stylesheet" href="/css/main.css">



<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous">

<script class="next-config" data-name="main" type="application/json">{"hostname":"hc1023.github.io","root":"/","images":"/images","scheme":"Gemini","darkmode":false,"version":"8.9.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12},"copycode":false,"bookmark":{"enable":false,"color":"#222","save":"auto"},"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"stickytabs":false,"motion":{"enable":false,"async":false,"transition":{"post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"prism":false,"i18n":{"placeholder":"Searching...","empty":"We didn't find any results for the search: ${query}","hits_time":"${hits} results found in ${time} ms","hits":"${hits} results found"},"path":"/search.xml","localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":true}}</script><script src="/js/config.js"></script>
<meta name="description" content="Review of alignment and SNP calling algorithms for next-generation sequencing data2015, J Appl Genetics  Bioinformatic tools for next-generation sequencing (NGS) data processing. Two of most significa">
<meta property="og:type" content="article">
<meta property="og:title" content="Reading--alignment and SNP calling algorithms">
<meta property="og:url" content="https://hc1023.github.io/2020/05/19/reading-alignment-and-SNP-calling-algorithms/index.html">
<meta property="og:site_name" content="Hc&#39;s Blog">
<meta property="og:description" content="Review of alignment and SNP calling algorithms for next-generation sequencing data2015, J Appl Genetics  Bioinformatic tools for next-generation sequencing (NGS) data processing. Two of most significa">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2020-05-19T00:56:15.000Z">
<meta property="article:modified_time" content="2020-05-22T01:45:57.000Z">
<meta property="article:author" content="Huang Sisi">
<meta property="article:tag" content="reading">
<meta name="twitter:card" content="summary">


<link rel="canonical" href="https://hc1023.github.io/2020/05/19/reading-alignment-and-SNP-calling-algorithms/">



<script class="next-config" data-name="page" type="application/json">{"sidebar":"","isHome":false,"isPost":true,"lang":"en","comments":true,"permalink":"https://hc1023.github.io/2020/05/19/reading-alignment-and-SNP-calling-algorithms/","path":"2020/05/19/reading-alignment-and-SNP-calling-algorithms/","title":"Reading--alignment and SNP calling algorithms"}</script>

<script class="next-config" data-name="calendar" type="application/json">""</script>
<title>Reading--alignment and SNP calling algorithms | Hc's Blog</title>
  




  <noscript>
    <link rel="stylesheet" href="/css/noscript.css">
  </noscript>
</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="headband"></div>

  <main class="main">
    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar" role="button">
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <p class="site-title">Hc's Blog</p>
      <i class="logo-line"></i>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>



<nav class="site-nav">
  <ul class="main-menu menu">
        <li class="menu-item menu-item-home"><a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a></li>
        <li class="menu-item menu-item-about"><a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a></li>
        <li class="menu-item menu-item-tags"><a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a></li>
        <li class="menu-item menu-item-categories"><a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>Categories</a></li>
        <li class="menu-item menu-item-archives"><a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a></li>
        <li class="menu-item menu-item-sitemap"><a href="/sitemap.xml" rel="section"><i class="fa fa-sitemap fa-fw"></i>Sitemap</a></li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup"><div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off" maxlength="80"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close" role="button">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div class="search-result-container no-result">
  <div class="search-result-icon">
    <i class="fa fa-spinner fa-pulse fa-5x"></i>
  </div>
</div>

    </div>
  </div>

</div>
        
  
  <div class="toggle sidebar-toggle" role="button">
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
  </div>

  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <div class="sidebar-panel-container">
        <!--noindex-->
        <div class="post-toc-wrap sidebar-panel">
            <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#Review-of-alignment-and-SNP-calling-algorithms-for-next-generation-sequencing-data"><span class="nav-number">1.</span> <span class="nav-text">Review of alignment and SNP calling algorithms for next-generation sequencing data</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Introduciton"><span class="nav-number">2.</span> <span class="nav-text">Introduciton</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Alignment-to-the-reference-genome"><span class="nav-number">3.</span> <span class="nav-text">Alignment to the reference genome</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Suffix-prefix-tries"><span class="nav-number">3.1.</span> <span class="nav-text">Suffix&#x2F;prefix tries</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Hash-tables"><span class="nav-number">3.2.</span> <span class="nav-text">Hash tables</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Alignment"><span class="nav-number">3.3.</span> <span class="nav-text">Alignment</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Post-alignment-processing"><span class="nav-number">3.4.</span> <span class="nav-text">Post-alignment processing</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#SNP-calling"><span class="nav-number">4.</span> <span class="nav-text">SNP calling</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Heuristic-methods"><span class="nav-number">4.1.</span> <span class="nav-text">Heuristic methods</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Probabilistic-methods"><span class="nav-number">4.2.</span> <span class="nav-text">Probabilistic methods</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Single-vd-multi-sample-SNP-calling"><span class="nav-number">4.3.</span> <span class="nav-text">Single- vd. multi-sample SNP calling</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Technical-issues"><span class="nav-number">5.</span> <span class="nav-text">Technical issues</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#Data-formats"><span class="nav-number">5.1.</span> <span class="nav-text">Data formats</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Computing-platforms-and-operating-systems"><span class="nav-number">5.2.</span> <span class="nav-text">Computing platforms and operating systems</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Notes"><span class="nav-number">6.</span> <span class="nav-text">Notes</span></a></li></ol></div>
        </div>
        <!--/noindex-->

        <div class="site-overview-wrap sidebar-panel">
          <div class="site-author site-overview-item animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Huang Sisi"
      src="/images/avatar.jpeg">
  <p class="site-author-name" itemprop="name">Huang Sisi</p>
  <div class="site-description" itemprop="description">No boundaries, no restrictions</div>
</div>
<div class="site-state-wrap site-overview-item animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
        <a href="/archives/">
          <span class="site-state-item-count">80</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
          <a href="/categories/">
        <span class="site-state-item-count">2</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
          <a href="/tags/">
        <span class="site-state-item-count">14</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>



        </div>
      </div>
    </div>
  </aside>
  <div class="sidebar-dimmer"></div>


    </header>

    
  <div class="back-to-top" role="button" aria-label="Back to top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>


    <div class="main-inner post posts-expand">


  


<div class="post-block">
  
  

  <article itemscope itemtype="http://schema.org/Article" class="post-content" lang="en">
    <link itemprop="mainEntityOfPage" href="https://hc1023.github.io/2020/05/19/reading-alignment-and-SNP-calling-algorithms/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.jpeg">
      <meta itemprop="name" content="Huang Sisi">
      <meta itemprop="description" content="No boundaries, no restrictions">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Hc's Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          Reading--alignment and SNP calling algorithms
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">Posted on</span>

      <time title="Created: 2020-05-19 08:56:15" itemprop="dateCreated datePublished" datetime="2020-05-19T08:56:15+08:00">2020-05-19</time>
    </span>
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar-check"></i>
      </span>
      <span class="post-meta-item-text">Edited on</span>
      <time title="Modified: 2020-05-22 09:45:57" itemprop="dateModified" datetime="2020-05-22T09:45:57+08:00">2020-05-22</time>
    </span>

  
</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">
        <h2 id="Review-of-alignment-and-SNP-calling-algorithms-for-next-generation-sequencing-data"><a href="#Review-of-alignment-and-SNP-calling-algorithms-for-next-generation-sequencing-data" class="headerlink" title="Review of alignment and SNP calling algorithms for next-generation sequencing data"></a>Review of alignment and SNP calling algorithms for next-generation sequencing data</h2><p><em>2015, J Appl Genetics</em></p>
<ul>
<li>Bioinformatic tools for next-generation sequencing (NGS) data processing.</li>
<li>Two of most significant tasks: <ul>
<li><strong>alignment to a reference genome -&gt; suffix tries and hash tables.</strong> Suffix array-based aligners are memory efficient and work faster than hash-based aligners, but they are less accurate. Hash table algorithms tend to be slower but more sensitive.</li>
<li><strong>detection of single nucleotide polymorphisms (SNPs) -&gt; heuristic and probabilistic methods.</strong> Due to the computational demands of heuristic methods, probabilistic methods are more commonly used.</li>
</ul>
</li>
</ul>
<span id="more"></span>

<h2 id="Introduciton"><a href="#Introduciton" class="headerlink" title="Introduciton"></a>Introduciton</h2><p>Existing NGS technologies provide us with a throughput that is at least 100 times larger than that obtained by Sanger sequencing, and the technologies are still improving. It led to a tremendous increase of sequencing speed and thus obtaining complete data of any species within a short period.<br>A major advantage of whole genome sequencing (WGS) stems from the potential to detect genetic variants which mark individual diviations from the reference genome. These are not only SNPs, but also small insertions/deletions (InDels), as well as large structual variants (SV), e.g. copy number variations (CNVs).<br>DIFFICULTIES: NGS represents a throuput technology, so it is highly sensitive to technological errors, consequently making the process of utilising NGS data for research highly dependent on reliable bioinformatics tools. Two major steps: alignment and detection of SNPs.</p>
<h2 id="Alignment-to-the-reference-genome"><a href="#Alignment-to-the-reference-genome" class="headerlink" title="Alignment to the reference genome"></a>Alignment to the reference genome</h2><p>First computaional step in NGS analysis: alignment of short reads to the existing reference genome or, alternatively, a de novo assembly without a reference.</p>
<p><strong>Indexing before alignment</strong>: most aligners construct indices for the reference sequence and/or for the short sequenced read data set. Indexing the reference genome is most common because it is andvantageous in terms of computational time.<br>Two main <strong>indexing algorithms</strong>: suffix/prefix tries or hash tables incoporated in vast majority of software tools, but exceptions exist, e.g. Slider sofware adopts merge-sorting.</p>
<h3 id="Suffix-prefix-tries"><a href="#Suffix-prefix-tries" class="headerlink" title="Suffix/prefix tries"></a>Suffix/prefix tries</h3><ul>
<li>suffix/prefix trie</li>
</ul>
<p>A data structure representing the set of suffixes of a given string of characters, enabling fast match of the string, which in our case represents a sequence read to the reference genome. However, it is not efficient enough, resulting in a very long computing time for very large NGS data sets.</p>
<ul>
<li>enhanced suffix array</li>
</ul>
<p>More efficient and enables storing large genomes in computer memory. It is composed of a suffix array and some auxiliary arrays.</p>
<ul>
<li>FM-index</li>
</ul>
<p>About BWT and FM-index, see <a href="https://hc1023.github.io/2020/03/17/Short-Read-Alignment/">Hc’s Blog: Short Read Alignment</a>.</p>
<p>Combine the Burrows-Wheeler transform (BWT) algorithm with auxiliary data structures. A suffix array created from BWT altered sequences is significantly more efficient than that from primary sequence. The most popular software tools implementing the FM-index algorithm with BWT are BWA, Bowtie, Bowtie2, SOAP3 and SOAP3-dp (2-BWT, with some modification).</p>
<h3 id="Hash-tables"><a href="#Hash-tables" class="headerlink" title="Hash tables"></a>Hash tables</h3><p>About basic conception of Hashing, see <a href="https://hc1023.github.io/2019/12/31/Hashing/">Hc’s Blog: Hashing</a>.</p>
<p>Seed-and-extend method: divide a query sequence into words called k-mers and the position of each k-mer is kept in the hash table. k-mers are sought in another sequence and their exact matches are termed seeds. Afterwards, seeds are extended and joined without gaps and then their alignment is refined using the Smith-Waterman alignment algorithm.</p>
<ul>
<li>spaced seed</li>
<li>q-gram filter and multiple seed hits</li>
<li>seed extension</li>
</ul>
<p>First two rely on the fast lookup in a hash table, while the seed extension algorithm is based on accelerating the standard Smith-Waterman alignment algorithm.</p>
<h3 id="Alignment"><a href="#Alignment" class="headerlink" title="Alignment"></a>Alignment</h3><p>Regardless of the indexing method, the actual alignment is performed using either the Smith-Waterman or the Needle-Wunsch algorithms.<br>Alignment gaps usually result from small-scale genome rearrangements, such as InDels. Thus, allowing for gaps in the alignment is preferred.<br>And currently, all of the commonly use alignment programs support a paired-end alignment mode, in which reads sequenced in both the forward and reverse orders are considered simultaneously as a pair, providing a longer piece of sequence to be aligned and, thus, significantly inproving the quality of alignment. See <a href="https://hc1023.github.io/2020/03/08/paired-end-sequencing/">Hc’s Blog: Illumina Sequencing (paired-end)</a>.</p>
<h3 id="Post-alignment-processing"><a href="#Post-alignment-processing" class="headerlink" title="Post-alignment processing"></a>Post-alignment processing</h3><ul>
<li>output file format converting</li>
</ul>
<p>Reduction of output data set size and guaranteen downstream compatibility with variant callers.</p>
<ul>
<li>create reports from the alignment process</li>
</ul>
<p>A simple summary including the total number of aligned reads, the number of properly aligned read pairs of reads or the number of reads aligned exactly once.</p>
<ul>
<li>remove polymerase chain reaction (PCR) artefacts</li>
</ul>
<p>The same read pairs occuring many times in the raw data and changing the estimate of the true coverage of the reference genome. (samtools rmdup tool is an example)</p>
<h2 id="SNP-calling"><a href="#SNP-calling" class="headerlink" title="SNP calling"></a>SNP calling</h2><p>Pre-variant calling process involve alignment artefact correction and sequence quality score recalibration. And for a reliable variant detection, a high depth of coverage is a prerequisite, since a high number of reads aligned to each base is used to differentiate between sequencing errors and true polymorphisms.<br>SNP calling may be defined as the process of identifying sites differing from a reference sequence, while genotype calling refers to the estimation of genotypes. They are based on either heuristic or probabilistic methods.</p>
<h3 id="Heuristic-methods"><a href="#Heuristic-methods" class="headerlink" title="Heuristic methods"></a>Heuristic methods</h3><ul>
<li>Based on multiple information sources associated with the structure and quality of data. </li>
<li>High computational demands -&gt; less commonly used.</li>
<li>For variant detection, a heuristic algorithm determineds the genotype based on the thresholds for coverage (minimum 33),  base quality (minimum 20) and variant allele frequency (minimum 0.08).</li>
<li>Followed by statistical test. Fisher’s exact test of read counts supporting each allele is applied compared to the expected distribution based on sequencing error alone.</li>
</ul>
<h3 id="Probabilistic-methods"><a href="#Probabilistic-methods" class="headerlink" title="Probabilistic methods"></a>Probabilistic methods</h3><ul>
<li>Based on genotype likelihood calculations and adopts Bayes’ theorem.</li>
<li>Provide measures of statistical uncertainty for called genotypes, making it possible to monitor the accuracy of genotype calling.</li>
<li>Additional information concerning allele frequencies and linkage disequilibrium patterns may be included in the analysis.</li>
</ul>
<p>In the Bayesian framework, the computed likelihood is combined with a prior genotype probability, which leads to a posterior probability of a genotype. As a result, the genotype with the highest posterior probability is chosen. The ratio between the highest and the second highest probabilities may be used as a measure of confidence.<br>A uniform prior probability may be selected as equal for all genotypes; alternatively, a non-uniform prior can be pre-imposed based on additional information, such as dbSNP (SNP database) entries, the reference sequence structures or features provided by the analysed sample.</p>
<h3 id="Single-vd-multi-sample-SNP-calling"><a href="#Single-vd-multi-sample-SNP-calling" class="headerlink" title="Single- vd. multi-sample SNP calling"></a>Single- vd. multi-sample SNP calling</h3><ul>
<li>Single</li>
</ul>
<p>Advantage: fast.<br>Disadvange: difficult to draw inferences on a population-wide level and requires custom written programs.</p>
<ul>
<li>Multi </li>
</ul>
<p>Advantage: priors may be improved based on the information on allele frequency and on verification of whether the obtained genotypes follow Hardy-Weinberg equilibrium; the latter is only useful if unrelated individuals are considered.<br>Disadvantage: much more CPU time- and resource- consuming than individual variant calling.</p>
<h2 id="Technical-issues"><a href="#Technical-issues" class="headerlink" title="Technical issues"></a>Technical issues</h2><h3 id="Data-formats"><a href="#Data-formats" class="headerlink" title="Data formats"></a>Data formats</h3><table>
<thead>
<tr>
<th>Format</th>
<th>Data</th>
</tr>
</thead>
<tbody><tr>
<td>FASTQ</td>
<td>short sequence reads</td>
</tr>
<tr>
<td>FASTA</td>
<td>reference genomes</td>
</tr>
<tr>
<td>SAM</td>
<td>output of alignment</td>
</tr>
<tr>
<td>BAM</td>
<td>binaray file of SAM</td>
</tr>
<tr>
<td>VCF</td>
<td>output of variant calling</td>
</tr>
</tbody></table>
<h3 id="Computing-platforms-and-operating-systems"><a href="#Computing-platforms-and-operating-systems" class="headerlink" title="Computing platforms and operating systems"></a>Computing platforms and operating systems</h3><p>With the decrease in sequencing costs, data processing costs are rising, while storing and processing very large amounts of data become a challenge.</p>
<ul>
<li>large data sets -&gt; memory requirements.</li>
<li>computationally demanding -&gt; supercomputing infrastructure involving many CPUs.</li>
<li>accelerate NGS processing -&gt; multi-threaded mode.</li>
</ul>
<h2 id="Notes"><a href="#Notes" class="headerlink" title="Notes"></a>Notes</h2><p>For less experienced users, SAMtools’ mpileup and bcftools are recommended because their ease of use. On the other hand, GATK, which is provided with a detailed user manual, is robust towards data artifacts. Nevertheless, both programs are frequently updated, provide detailed log files describing the process of SNP calling and the agreement between their outputs is high.</p>

    </div>

    
    
    

    <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/reading/" rel="tag"># reading</a>
          </div>

        

          <div class="post-nav">
            <div class="post-nav-item">
                <a href="/2020/05/16/tex-spell-check/" rel="prev" title="Tex语法检查">
                  <i class="fa fa-chevron-left"></i> Tex语法检查
                </a>
            </div>
            <div class="post-nav-item">
                <a href="/2020/06/03/Algebra-notes/" rel="next" title="表示论 Algebra notes">
                  表示论 Algebra notes <i class="fa fa-chevron-right"></i>
                </a>
            </div>
          </div>
    </footer>
  </article>
</div>






</div>
  </main>

  <footer class="footer">
    <div class="footer-inner">


<div class="copyright">
  &copy; 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Huang Sisi</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.js.org/" rel="noopener" target="_blank">NexT.Gemini</a>
  </div>

    </div>
  </footer>

  
  <script src="https://cdn.jsdelivr.net/npm/animejs@3.2.1/lib/anime.min.js" integrity="sha256-XL2inqUJaslATFnHdJOi9GfQ60on8Wx1C2H8DYiN1xY=" crossorigin="anonymous"></script>
<script src="/js/comments.js"></script><script src="/js/utils.js"></script><script src="/js/next-boot.js"></script>

  
<script src="https://cdn.jsdelivr.net/npm/hexo-generator-searchdb@1.4.0/dist/search.js" integrity="sha256-vXZMYLEqsROAXkEw93GGIvaB2ab+QW6w3+1ahD9nXXA=" crossorigin="anonymous"></script>
<script src="/js/third-party/search/local-search.js"></script>





  




  

  <script class="next-config" data-name="enableMath" type="application/json">true</script><script class="next-config" data-name="mathjax" type="application/json">{"enable":true,"tags":"none","js":{"url":"https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js","integrity":"sha256-r+3itOMtGGjap0x+10hu6jW/gZCzxHsoKrOd7gyRSGY="}}</script>
<script src="/js/third-party/math/mathjax.js"></script>



</body>
</html>
